community/kernel/src/main/java/org/neo4j/kernel/impl/locking/Locks.java

```diff
@@ -107,6 +109,9 @@ public Factory( String key, String... altKeys )
         /** Release all locks. */
         void releaseAll();
 
+        /** Associate with transaction. */
+        void setTx( Transaction tx );
```

Comment:
Cringe no more, I'll change it! Originally I was binding it directly to a thread, but transactions it seems can suspend/resume on another thread, so changed it to that... but with transactions a setter doesn't make the best sense anymore. I don't think there are any issues with changing it.